### PR TITLE
fix: switch dropdown component in table pagination to native select

### LINKED ISSUE #393: The displayed option in 'Rows per page' is not highlighted in the drop down list
**Describe the bug**
The displayed option in 'Rows per page' is not highlighted in the drop down list

**To Reproduce**
1. Go to 'https://storybook-preview-319--upbeat-sinoussi-f675aa.netlify.app/?path=/story/fileuploader--csv'
2. Click on 'Browse Files'
3. Upload file
4. Click on 'Rows per page' drop down
5. Select any option ( eg:100)
6. Verify the selected option is displayed 
7. Click on the drop down 
8. Notice that the displayed option (100) is not selected but option '10' is

**Expected behavior**
The displayed option in 'Rows per page' is not highlighted in the drop down list

**Screenshots**
https://screencast-o-matic.com/watch/cr6XrIVXOWd

**Desktop (please complete the following information):**
OS: [Win 10]
Browser [chrome,]
Version [Version 94.0.4606.81 (64-bit)]

**Additional context**
[gradebookExport (4).csv](https://github.com/cengage/react-magma/files/7340559/gradebookExport.4.csv)

### PR DESCRIPTION
**Pull request by greenervigil**
switch out the dropdown component use in the table component to the native select component to fix focus state issue

closes #393

Issue: #393 

## Checklist 
- [x] changeset has been added
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by ccedrone — APPROVED** [review]

File: packages/react-magma-dom/src/components/Table/TablePagination.tsx [hunk 1]
```diff
@@ -11,24 +11,12 @@ import { useIsInverse } from '../../inverse';
...
 import { ButtonGroup, ButtonGroupAlignment } from '../ButtonGroup';
+import { NativeSelect } from '../NativeSelect';
 
 export interface BaseTablePaginationProps
   extends React.HTMLAttributes<HTMLDivElement> {
-  /**
-   * Position of the dropdown content
-   * @default DropdownDropDirection.up
-   */
-  dropdownDropDirection?: DropdownDropDirection;
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/TablePagination.tsx:31]: Could this be considered a breaking change, now that we've removed this prop?

**Comment by silvalaura** [comment]
I think if we are going to wait to merge this until after our next patch/minor release, we should open a separate PR that we do merge immediately that marks `dropdownDropDirection` as deprecated. That way, we will have "warned" consumers about the incoming breaking change before we introduce it. Thoughts?

**Comment by ccedrone** [comment]
We did the same thing with `button` colors I believe right? V3 required us to drop a couple of them and we documented the change, seemed fine to me.

**Review by orion-cengage — APPROVED** [review]

**Review by ccedrone — APPROVED** [review]

**Comment by greenervigil** [comment]
Could not get the Base Pagination props to render in the documentation.

File: website/react-magma-docs/src/pages/api/table.mdx [hunk 2]
```diff
@@ -1064,20 +1062,16 @@ All of the [standard td attributes](https://developer.mozilla.org/en-US/docs/Web
...
 
-**This component uses `forwardRef`. The ref is applied to the Alert container `div` element.**
+**This component uses `forwardRef`. The ref is applied to the container.**
 
 All of the [global HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes) can be provided as props and will be applied to the wrapping `div` element that gets rendered.
 
-### Common Props
-
-<BaseTablePaginationProps />
-
```
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/table.mdx:1074]: `BaseTablePaginationProps` were deleted in the mdx file, maybe that's why they aren't showing up?

File: .changeset/light-dancers-explode.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'react-magma-dom': major
+---
+
+Breaking Change: (TablePagination, DataGrid) switch dropdown component in table pagination to native select
```
Comment (silvalaura) [inline .changeset/light-dancers-explode.md:5]: This is now a minor change, without a breaking change. So we can update it to

```suggestion
---
'react-magma-dom': minor
---

(TablePagination, DataGrid): Switch dropdown component in table pagination to native select
Deprecations: `dropdownDropDirection` prop is now deprecated
```

File: packages/react-magma-dom/src/components/Table/TablePagination.tsx [hunk 4]
```diff
@@ -135,10 +130,10 @@ export const TablePagination = React.forwardRef<
   TablePaginationProps
 >((props, ref) => {
   const {
+    dropdownDropDirection,
     testId,
     defaultPage,
     defaultRowsPerPage = 10,
-    dropdownDropDirection = DropdownDropDirection.up,
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Table/TablePagination.tsx:141]: Should we keep this default in the meantime?

Comment (greenervigil) [reply packages/react-magma-dom/src/components/Table/TablePagination.tsx:141]: I think you would want the Native Select to be the default option and if we provide a default value to dropdownDropDirection then the Dropdown Component would be used.

**Review by silvalaura — APPROVED** [review]
✨

**Pull request merged**